client/views/admin/emailInbox/EmailInboxForm.js

```diff
@@ -0,0 +1,361 @@
+		imapSecure: imap.secure ?? false,
+	};
+};
+
+export function EmailInboxEditWithData({ id }) {
+	const t = useTranslation();
+	const { value: data, error, phase: state } = useEndpointData(`email-inbox/${ id }`);
+
+	if ([state].includes(AsyncStatePhase.LOADING)) {
+		return <FormSkeleton/>;
+	}
+
+	if (error || !data) {
+		return <Box mbs='x16'>{t('EmailInbox_not_found')}</Box>;
+	}
+
+	return <EmailInboxForm id={id} data={data} />;
+}
+
+export default function EmailInboxForm({ id, data }) {
+	const t = useTranslation();
+	const dispatchToastMessage = useToastMessageDispatch();
+	const setModal = useSetModal();
+	const [emailError, setEmailError] = useState();
+	const { values, handlers, hasUnsavedChanges } = useForm(getInitialValues(data));
+
+	const {
+		handleActive,
+		handleName,
+		handleEmail,
+		handleDescription,
+		handleSenderInfo,
+		handleDepartment,
+		// SMTP
+		handleSmtpServer,
+		handleSmtpPort,
+		handleSmtpUsername,
+		handleSmtpPassword,
+		handleSmtpSecure,
+		// IMAP
+		handleImapServer,
+		handleImapPort,
+		handleImapUsername,
+		handleImapPassword,
+		handleImapSecure,
+	} = handlers;
+	const {
+		active,
+		name,
+		email,
+		description,
+		senderInfo,
+		department,
+		// SMTP
+		smtpServer,
+		smtpPort,
+		smtpUsername,
+		smtpPassword,
+		smtpSecure,
+		// IMAP
+		imapServer,
+		imapPort,
+		imapUsername,
+		imapPassword,
+		imapSecure,
+	} = values;
+
+	const router = useRoute('admin-email-inboxes');
+
+	const close = useCallback(() => router.push({}), [router]);
+
+	const saveEmailInbox = useEndpointAction('POST', 'email-inbox');
+	const deleteAction = useEndpointAction('DELETE', `email-inbox/${ id }`);
+	const emailAlreadyExistsAction = useEndpointAction('GET', `email-inbox.search?email=${ email }`);
+
+	useComponentDidUpdate(() => {
+		setEmailError(!isEmail(email) ? t('Validate_email_address') : null);
+	}, [t, email]);
+	useComponentDidUpdate(() => {
+		!email && setEmailError(null);
+	}, [email]);
+
+	const handleRemoveClick = useMutableCallback(async () => {
+		const result = await deleteAction();
+		if (result.success === true) {
+			close();
+		}
+	});
+
+	const handleDelete = useMutableCallback((e) => {
+		e.stopPropagation();
+		const onDeleteManager = async () => {
+			try {
+				await handleRemoveClick();
+				dispatchToastMessage({ type: 'success', message: t('Removed') });
+			} catch (error) {
+				dispatchToastMessage({ type: 'error', message: error });
+			}
+			setModal();
+		};
+
+		setModal(<DeleteWarningModal
+			onDelete={onDeleteManager}
+			onCancel={() => setModal()}
+		>{t('You_will_not_be_able_to_recover_email_inbox')}</DeleteWarningModal>);
+	});
+
+	const handleSave = useMutableCallback(async () => {
+		const smtp = { server: smtpServer, port: parseInt(smtpPort), username: smtpUsername, password: smtpPassword, secure: smtpSecure };
+		const imap = { server: imapServer, port: parseInt(imapPort), username: imapUsername, password: imapPassword, secure: imapSecure };
```

Comment:
```suggestion
		const smtp = { server: smtpServer, port: smtpPort, username: smtpUsername, password: smtpPassword, secure: smtpSecure };
		const imap = { server: imapServer, port: imapPort, username: imapUsername, password: imapPassword, secure: imapSecure };
```

Reply:
We cannot remove that because `TextInput` component, when you type something, the value is being changed to `string`.